benchmarks/p256.cc

```diff
@@ -94,15 +103,17 @@ P256_SHA256_ECDSA_Verify(benchmark::State& state)
   bytes nonce = hex_to_bytes(
     "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA");
 
-  assert(Hacl_P256_ecdsa_sign_p256_sha2(
-    signature.data(), msg.size(), msg.data(), sk.data(), nonce.data()));
+  if (!Hacl_P256_ecdsa_sign_p256_sha2(
+        signature.data(), msg.size(), msg.data(), sk.data(), nonce.data())) {
+    state.SkipWithError("Error signing");
+    return;
+  }
 
+  bytes r(signature.begin(), signature.begin() + 32);
```

Comment:
Since `asserts` are stripped in `--release` it's generally good we don't use them in benchmarks. So, from all the changes, I assume only this one could be important? It doesn't seem to change much on my machine, though:

```
--------------------------------------------------------------------
Benchmark                          Time             CPU   Iterations
--------------------------------------------------------------------

--- With r,s before loop
P256_SHA256_ECDSA_Sign        328377 ns       328279 ns         2130
P256_SHA256_ECDSA_Verify      734751 ns       734636 ns          947
P256_ECDH                     320494 ns       320433 ns         2184
Openssl_P256_ECDSA_Sign        20389 ns        20383 ns        34030
Openssl_P256_ECDSA_Verify      54693 ns        54673 ns        12692
Openssl_P256_ECDH              38962 ns        38950 ns        17927

--- With r,s in loop
P256_SHA256_ECDSA_Sign        325494 ns       325416 ns         2150
P256_SHA256_ECDSA_Verify      730341 ns       730129 ns          952
P256_ECDH                     318984 ns       318906 ns         2192
Openssl_P256_ECDSA_Sign        20162 ns        20154 ns        34424
Openssl_P256_ECDSA_Verify      54081 ns        54060 ns        12041
Openssl_P256_ECDH              38737 ns        38727 ns        18051
```

Do you see a difference on your machine?

Reply:
Sanity: On `main`. Huch?

```
P256_SHA256_ECDSA_Sign        325763 ns       325648 ns         2148
P256_SHA256_ECDSA_Verify         507 ns          507 ns      1370639
P256_ECDH                        483 ns          483 ns      1452764
Openssl_P256_ECDSA_Sign        20378 ns        20371 ns        34618
Openssl_P256_ECDSA_Verify      54737 ns        54717 ns        12710
Openssl_P256_ECDH              38831 ns        38820 ns        17951
```

Reply:
I don't see any changes in performance from this change either. It's dominated by the math.